### PR TITLE
Convert some skill packets to structure format

### PR DESCRIPTION
**Pull request by Asheraf**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/packets_struct.h [hunk 1]
```diff
@@ -3984,6 +3984,27 @@ struct PACKET_ZC_NOTIFY_SKILL {
...
+#else
+struct PACKET_ZC_USE_SKILL {
+	int16 PacketType;
+	uint16 SKID;
+	int32 level;
+	uint32 targetAID;
+	uint32 srcAID;
+	int8 result;
+} __attribute__((packed));
+DEFINE_PACKET_HEADER(ZC_USE_SKILL, 0x09cb);
```
Comment (4144) [inline src/map/packets_struct.h:4006]: this packet can be used from
2013-07-31cRagexe
2013-07-24eRagexeRE

File: src/map/packets_struct.h [hunk 2]
```diff
@@ -4017,6 +4017,15 @@ struct PACKET_ZC_NOTIFY_GROUNDSKILL {
 } __attribute__((packed));
 DEFINE_PACKET_HEADER(ZC_NOTIFY_GROUNDSKILL, 0x0117);
 
+#if PACKETVER >= 20081112
+struct PACKET_ZC_SKILL_POSTDELAY {
+	int16 PacketType;
+	uint16 SKID;
+	uint32 DelayTM;
+} __attribute__((packed));
+DEFINE_PACKET_HEADER(ZC_SKILL_POSTDELAY, 0x043d);
```
Comment (4144) [inline src/map/packets_struct.h:4026]: this packet can be used from
2008-11-12aRagexe
2008-11-11aRagexeRE

**Review by 4144 — CHANGES_REQUESTED** [review]

File: src/map/clif.c [hunk 3]
```diff
@@ -5623,98 +5625,60 @@ static void clif_skill_cooldown(struct map_session_data *sd, uint16 skill_id, un
...
 	nullpo_ret(dst);
 
-	damage = (int)cap_value(in_damage, INT_MIN, INT_MAX);
+	int damage = (int)cap_value(in_damage, INT_MIN, INT_MAX);
 	type = clif_calc_delay(type, div, damage, ddelay);
 
 #if PACKETVER >= 20131223
-	if (type == BDT_SKILL) type = BDT_MULTIHIT; //bugreport:8263
+	// For some reason, late 2013 and newer clients have
+	// a issue that causes players and monsters to endure
```
Comment (EPuncker) [inline src/map/clif.c:5636]:
```suggestion
	// an issue that causes players and monsters to endure
```

File: src/map/clif.c [hunk 4]
```diff
@@ -5689,52 +5689,50 @@ static int clif_skill_damage(struct block_list *src, struct block_list *dst, int
...
+	type = (type > 0) ? type : skill->get_hit(skill_id, skill_lv);
+	type = clif_calc_delay(type, div, damage, ddelay);
 
-	if(sc && sc->count) {
-		if(sc->data[SC_ILLUSION] && damage)
-			damage = damage*(sc->data[SC_ILLUSION]->val2) + rnd()%100;
+	const struct status_change *sc = status->get_sc(dst);
+	if (sc != NULL && sc->count) {
+		if (sc->data[SC_ILLUSION] != NULL && damage != 0)
+			damage *= (sc->data[SC_ILLUSION]->val2) + rnd() % 100;
```
Comment (MishimaHaruna) [inline src/map/clif.c:5701]: This isn't equivalent to the existing code: it used to be `damage = damage * val2 + rnd` while now it's `damage = damage * (val2 + rnd)`. Can you double check what it's meant to be like?

File: src/map/clif.c [hunk 5]
```diff
@@ -5623,98 +5623,60 @@ static void clif_skill_cooldown(struct map_session_data *sd, uint16 skill_id, un
...
+		type = BDT_MULTIHIT; //bugreport:8263
 #endif
 
-	if ((sc = status->get_sc(dst)) && sc->count) {
-		if (sc->data[SC_ILLUSION] && damage)
-			damage = damage * (sc->data[SC_ILLUSION]->val2) + rnd() % 100;
+	const struct status_change *sc = status->get_sc(dst);
+	if (sc != NULL && sc->count) {
+		if (sc->data[SC_ILLUSION] != NULL && damage != 0)
+			damage *= (sc->data[SC_ILLUSION]->val2) + rnd() % 100;
```
Comment (MishimaHaruna) [inline src/map/clif.c:5646]: This isn't equivalent to the existing code: it used to be `damage = damage * val2 + rnd` while now it's `damage = damage * (val2 + rnd)`. Can you double check what it's meant to be like?

**Review by MishimaHaruna — CHANGES_REQUESTED** [review]

File: src/map/clif.c (hunk 4 above)
Comment (Asheraf) [reply src/map/clif.c]: reverted the change

File: src/map/clif.c (hunk 5 above)
Comment (Asheraf) [reply src/map/clif.c]: reverted the change

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**